mithril-aggregator/src/database/provider/open_message.rs

```diff
@@ -0,0 +1,481 @@
+use mithril_common::StdError;
+
+use mithril_common::{
+    entities::{Beacon, Epoch, SignedEntityType},
+    sqlite::{HydrationError, Projection, SqLiteEntity, WhereCondition},
+    sqlite::{Provider, SourceAlias},
+};
+
+use chrono::NaiveDateTime;
+use sqlite::Row;
+use sqlite::{Connection, Value};
+
+use std::sync::Arc;
+
+use tokio::sync::Mutex;
+use uuid::Uuid;
+
+type StdResult<T> = Result<T, StdError>;
+
+/// ## OpenMessage
+///
+/// An open message is a message open for signatures. Every signer may send a
+/// single signature for this message from which a multi signature will be
+/// generated if possible.
+#[allow(dead_code)]
+pub struct OpenMessage {
+    /// OpenMessage unique identifier
+    open_message_id: Uuid,
+
+    /// Epoch
+    epoch: Epoch,
+
+    /// Beacon, this is the discriminant of this message type in the current
+    /// Epoch
+    beacon: Beacon,
+
+    /// Type of message
+    signed_entity_type: SignedEntityType,
+
+    /// Message content
+    message: String,
+
+    /// Message creation datetime, it is set by the database.
+    created_at: NaiveDateTime,
+}
+
+impl SqLiteEntity for OpenMessage {
+    fn hydrate(row: Row) -> Result<Self, HydrationError>
+    where
+        Self: Sized,
+    {
+        let open_message_id = row.get::<String, _>(0);
+        let open_message_id = Uuid::parse_str(&open_message_id).map_err(|e| {
+            HydrationError::InvalidData(format!(
+                "Invalid UUID in open_message.open_message_id: '{open_message_id}'. Error: {e}"
+            ))
+        })?;
+        let message = row.get::<String, _>(4);
+        let epoch_setting_id = row.get::<i64, _>(1);
+        let epoch_val = u64::try_from(epoch_setting_id)
+            .map_err(|e| panic!("Integer field open_message.epoch_setting_id (value={epoch_setting_id}) is incompatible with u64 Epoch representation. Error = {e}"))?;
+
+        let signed_entity_type_id = usize::try_from(row.get::<i64, _>(3)).map_err(|e| {
+            panic!(
+                "Integer field open_message.signed_entity_type_id cannot be turned into usize: {e}"
+            )
+        })?;
+        let signed_entity_type = SignedEntityType::from_repr(signed_entity_type_id)
+            .ok_or_else(|| HydrationError::InvalidData(format!(
+                "Field open_message.signed_type_id can be either 0, 1 or 2, ({signed_entity_type_id} given)."
```

Comment:
That's true but as long as there is no more entity types this will remain true and give a more helpful message to developers. A more automatic error message could be written, PR welcome ;)